src/Ombi.TheMovieDbApi/TheMovieDbApi.cs

```diff
@@ -99,65 +105,95 @@ public async Task<List<MovieSearchResult>> SimilarMovies(int movieId, string lan
         public async Task<MovieResponseDto> GetMovieInformationWithExtraInfo(int movieId, string langCode = "en")
         {
             var request = new Request($"movie/{movieId}", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("append_to_response", "videos,release_dates");
-            request.FullUri = request.FullUri.AddQueryParameter("language", langCode);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("append_to_response", "videos,release_dates");
+            request.AddQueryString("language", langCode);
             AddRetry(request);
             var result = await Api.Request<MovieResponse>(request);
             return Mapper.Map<MovieResponseDto>(result);
         }
 
-        public async Task<List<MovieSearchResult>> SearchMovie(string searchTerm, int? year, string langageCode)
+        public async Task<List<MovieSearchResult>> SearchMovie(string searchTerm, int? year, string langCode)
         {
             var request = new Request($"search/movie", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("query", searchTerm);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("query", searchTerm);
+            request.AddQueryString("language", langCode);
             if (year.HasValue && year.Value > 0)
             {
-                request.FullUri = request.FullUri.AddQueryParameter("year", year.Value.ToString());
+                request.AddQueryString("year", year.Value.ToString());
             }
+
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+
             AddRetry(request);
 
             var result =  await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> PopularMovies(string langageCode)
+        public async Task<List<MovieSearchResult>> PopularMovies(string langCode)
         {
-            var request = new Request($"movie/popular", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
```

Comment:
Can we add
```
request.AddQueryString("sort_by", "popularity.desc");
```

Reply:
@tidusjar Yes, I can add. I was torn between including it because it is more explicit or excluding it because it is the [default](https://developers.themoviedb.org/3/discover/movie-discover) and fewer bits across the wire.